libs/fluid-elements/button-group/src/lib/button-group/button-group.ts

```diff
@@ -0,0 +1,221 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  LitElement,
+  CSSResult,
+  css,
+  TemplateResult,
+  html,
+  property,
+  customElement,
+  unsafeCSS,
+} from 'lit-element';
+import { FluidButtonGroupItem } from '../button-group-item/button-group-item';
+import { FluidButtonGroupItemCheckedChangeEvent } from '../button-group-events';
+import { SPACE, ARROW_RIGHT, ARROW_LEFT } from '@dynatrace/shared/keycodes';
+import { FLUID_SPACING_X_SMALL } from '@dynatrace/fluid-design-tokens';
+import { getNextIndex } from '@dynatrace/shared/fluid-utils';
+
+let uniqueCounter = 0;
+
+/**
+ * A basic representation of the button-group wrapper for button-group-items
+ * @element fluid-button-group
+ * @slot - Default slot lets the user provide a label for the button-group.
+ * to user interaction.
+ */
+@customElement('fluid-button-group')
+export class FluidButtonGroup extends LitElement {
+  private _buttonGroupItems: FluidButtonGroupItem[] = [];
+
+  private _unique = `fluid-button-group-${uniqueCounter++}`;
+
+  /** Styles for the button-group component */
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        /**
+        * Legibility definitions should probably be
+        * shipped or imported from a core
+        */
+        -webkit-font-smoothing: antialiased;
+        -moz-osx-font-smoothing: grayscale;
+        text-rendering: optimizeLegibility;
```

Comment:
I don't like to ship this with every component! Furthermore, I want to have CSS processed by an autoprefixer to not have to write code that can be auto-generated.